### PR TITLE
fix(modal): restore broken stacked modals

### LINKED ISSUE #6103: 2.0.0 Nested modal not working correctly, backdrop in front and not closable
### Bug description:

Calling $uibModal.open from within an open modal dialog causes the new dialog being shown in the back of the active dialog. The screen front is covered by the related backdrop which makes the site unusable.

### Link to minimally-working plunker that reproduces the issue:

http://plnkr.co/edit/9cTZmaD0Veriujy9n7iE?p=preview
Click on "Open Me" and then any of the shown list items.
(Increase the size of the preview to see that the "lg" dialog is behind the active one)

### Version of Angular, UIBS, and Bootstrap

Angular:
1.5.x
UIBS:
2.0.0
Bootstrap:
3.3.6

### PR DESCRIPTION
**Pull request by fpipita**
Hi, 

with 2.0.0, stacked modals are not getting the `zIndex` property correctly set anymore because the `index` property needed by `ngStyle` to compute the `zIndex` property, is not available on the `ngStyle` scope (which is the modal scope).

This PR solves it by simply adding a new "private" property to the modal scope, `$$topModalIndex` which provides the actual modal index value.

Closes #6103

### CODE REVIEW

File: src/modal/modal.js [hunk 1]
```diff
@@ -521,6 +521,16 @@ angular.module('ui.bootstrap.modal', ['ui.bootstrap.stackedMap', 'ui.bootstrap.p
...
         appendToElement.addClass(modalBodyClass);
+        if (modal.scope) {
+          // ngStyle is associated with the modal scope and because the "index"
+          // property (which is private to the isolated scope of the uibModalWindow directive)
+          // is undefined here, it fails to compute the zIndex property.
+          // So we add it explicitly to the scope.
+          // This might need to be marked as a breaking change because
+          // "$$topModalIndex" could eventually shadow a property with the
+          // same name defined on the provided modal scope.
+          modal.scope.$$topModalIndex = topModalIndex;
```
Comment (icfantv) [inline src/modal/modal.js]: i'm ok w/ the name seeing as one is _not_ supposed to be using `$` as prefixes for their code.  but this is good insight to be aware of.

Comment (icfantv) [inline src/modal/modal.js]: i don't think it needs a BC because of that reason.

**Comment by icfantv** [comment]
@fpipita, i'm ok with this.  thanks for putting this together.

**Comment by fpipita** [comment]
@icfantv please, let me know whether I should remove the BC tag from the commit message.

**Comment by icfantv** [comment]
That's fine, or I can remove it on commit.

**Comment by fpipita** [comment]
Ok, done.

**Comment by icfantv** [comment]
@fpipita did you forget to push your changes?  i'm not seeing them.

**Comment by fpipita** [comment]
@icfantv I've pushed them, the BC tag is not in the commit body anymore. Did I get anything wrong?

**Comment by icfantv** [comment]
My bad, I was looking at the comment in the code.  Do you mind fixing that.

**Comment by fpipita** [comment]
No problem, I've cleaned it up.

**Comment by alejandromagnorsky** [comment]
@icfantv , do you mind making a new release (2.0.1) with this fix please? I'm waiting for it to publish a website that uses version 2.0.0 and has the stack modals bug.
Thank you.

**Comment by wesleycho** [comment]
Done.

**Pull request closed without merging**